tests/test_io.py

```diff
@@ -166,6 +256,16 @@ def test_upload_not_relative_to_base(monkeypatch, mock_urlopen):
     "upload_file_path", ["a/b/test-file.txt", Path("a/b/test-file.txt")]
 )
 def test_upload(mock_urlopen, upload_file_path):
+    """Test a basic call to the upload API.
```

Comment:
I am somewhat hesitant to add these docstring checks to the testing directories. I could see people writing fewer tests because they have to go through this with every new test they write and just don't want to deal with the hassle. I guess most tests usually don't take parameters in, so hopefully not too big of a deal and we can always remove the checks later if this does turn out to negatively impact test writing.

Reply:
Yeah, I can see your point there. And usually test function names are written in a way that makes it obvious what is being tested.

Having _some_ docstring there does make the code reference docs a bit nicer looking, if we care about that.

Reply:
I think we leave it in as you have it since you've done the work :) Then we can adjust the lint codes later if it is a problem.